dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -2456,4 +2475,6 @@ global using nfloat = global::System.Runtime.InteropServices.NFloat%3B
 	<Import Project="$(_TargetsDirectory)Xamarin.MacCatalyst.ObjCBinding.$(_ProjectLanguage).targets"  Condition="'$(_ProjectType)' == 'MacCatalystBindingProject' " />
 	
 	<Import Project="dotnet-xcsync.targets" />
+	<Import Project="Microsoft.MaciOS.Sdk.Xcode.targets" Condition=" $([MSBuild]::IsOSPlatform('osx')) " />
```

Comment:
This actually breaks the build on Windows, because we're trying to build the `_BuildXcodeProjects` target (it's been added to `BuildDependsOn` further above), but the `_BuildXcodeProjects` target can't be found (because `Microsoft.MaciOS.Sdk.Xcode.targets` isn't imported).

Sample binlog: [log-build-20240924_074321.binlog.zip](https://github.com/user-attachments/files/17116568/log-build-20240924_074321.binlog.zip)

Reply:
Let me move the conditions back into the targets